offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -447,11 +444,34 @@ class Offer {
     }
 
     private double calculateNetPrice() {
-        double netSum = 0.0
-        for (item in items) {
-            netSum += item.quantity * item.product.unitPrice
+        double netPrice
+        switch (selectedCustomerAffiliation.category) {
+            case AffiliationCategory.INTERNAL:
+                netPrice = calculateInternalNetPrice()
+                break
+            default:
+                netPrice = calculateExternalNetPrice()
+                break
         }
```

Comment:
```suggestion
        double netPrice = calculateExternalNetPrice()
       if (selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL){
                netPrice = calculateInternalNetPrice()
        }
```

Reply:
ok maybe having an else here would be better otherwise we trigger a calculation which is not needed

Reply:
we do not enter the default case so we do not trigger the calculation